### PR TITLE
[WIP] Support local xgboost regressor

### PR DESCRIPTION
**Pull request by rootfs**
the xgboost library PR #1041  needs to be merged first

### CODE REVIEW

File: pkg/model/estimator/local/lr_test.go [hunk 1]
```diff
@@ -243,7 +244,7 @@ var _ = Describe("Test LR Weight Unit", func() {
 			// Estimate Container Components Power using Linear Regression
 			modelWeightFilepath := config.GetDefaultPowerModelURL(types.DynPower.String(), types.ComponentEnergySource)
 			initModelURL := "https://raw.githubusercontent.com/sustainable-computing-io/kepler-model-db/main/models/v0.6/nx12/std_v0.6/rapl/DynPower/BPFOnly/SGDRegressorTrainer_1.json"
```
Comment (marceloamaral) [inline pkg/model/estimator/local/lr_test.go:246]: Probably need to update the model here, this model is SGD, not sure it it if compatible with XGBoost.
Did you try to locally?

**Review by marceloamaral — COMMENTED** [review]
Looks very promising!

File: pkg/model/estimator/local/lr_test.go (hunk 1 above)
Comment (marceloamaral) [reply pkg/model/estimator/local/lr_test.go:246]: Might be also good to change the name of the files and other references from `lr` to `xgb`.

**Comment by sunya-ch** [comment]
@rootfs I updated the xgboost to the main pipeline and upload the trained model on my local repo. Please use it first for testing the WIP code here.

The json of xgboost is in NormalizedNumericalFeature.Weight
You may redefine NormalizedNumericalFeature.Weight to interface{} instead of float and handle it separately for each type. 
Not sure but you may need to temporarily save the extracted Weight in json format can read it from the function `C.LoadModelFromJSON`. 

```yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: kepler-cfm
  namespace: kepler
data:
  MODEL_CONFIG: |
    NODE_TOTAL_INIT_URL =https://raw.githubusercontent.com/sunya-ch/kepler-model-db/trl-nx12/models/v0.7/nx12/std_v0.7/acpi/AbsPower/BPFOnly/XgboostFitTrainer_1.json
    NODE_COMPONENTS_INIT_URL=https://raw.githubusercontent.com/sunya-ch/kepler-model-db/trl-nx12/models/v0.7/nx12/std_v0.7/rapl/AbsPower/BPFOnly/XgboostFitTrainer_1.json
```

**Comment by rootfs** [comment]
the git history messed up. Close this PR and reopen in another clean branch

**Comment by rootfs** [comment]
replaced by #1052

**Pull request closed without merging**